glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -0,0 +1,524 @@
+        // <https://github.com/googlefonts/glyphsLib/blob/74c63244fdbef1da5/Lib/glyphsLib/filters/eraseOpenCorners.py#L105>
+        let t0_inv = 1.0 - t0;
+        if ((t0_inv < 0.5 && t1 < 0.5) || t0_inv < 0.3 || t1 < 0.3)
+            && t1 > 0.0001
+            && t0_inv > 0.0001
+        {
+            let ix = candidate.ix;
+            todo.extend([
+                (ix, SegmentOp::Replace(candidate.prev.subsegment(0.0..t0))),
+                (ix + 1, SegmentOp::Delete),
+                (
+                    ix + 2,
+                    SegmentOp::Replace(candidate.next.subsegment(t1..1.0)),
+                ),
+            ]);
+        }
+    }
+    if todo.is_empty() {
+        return None;
+    }
+
+    // it will be simpler to operate on segments:
+    let segments = kurbo::segments(path.elements().iter().copied()).collect::<Vec<_>>();
+    // and then we will make an equal number of ops as segments, also to simplify:
+    let mut ops = vec![SegmentOp::Retain; segments.len()];
+    for (ix, op) in todo {
+        // ix isn't always right because we didn't know the number of segments
+        // before, and we handle the first two points at the end
+        ops[ix % segments.len()] = op;
+    }
+
+    let mut els = Vec::with_capacity(path.elements().len());
+    let mut iter = segments
+        .iter()
+        .zip(ops)
+        .filter_map(|(seg, op)| op.apply(*seg))
+        .peekable();
+
+    // we will use `PathSeg::as_path_el` below, but that doesn't handle moveto
+    els.push(PathEl::MoveTo(iter.peek().unwrap().start()));
+    els.extend(iter.map(|seg| seg.as_path_el()));
+    if is_closed {
+        els.push(PathEl::ClosePath);
+    }
+    Some(BezPath::from_vec(els))
+}
+
+#[derive(Clone, Debug)]
+enum SegmentOp {
+    Retain,
+    Delete,
+    Replace(PathSeg),
+}
+
+impl SegmentOp {
+    fn apply(self, seg: PathSeg) -> Option<PathSeg> {
+        match self {
+            SegmentOp::Retain => Some(seg),
+            SegmentOp::Delete => None,
+            SegmentOp::Replace(seg) => Some(seg),
+        }
+    }
+}
+
+/// A potentially open corner.
+///
+/// This is any line segment, along with the preceding and following path segments.
+struct ErasureCandidate {
+    // index of the *first* participating segment, which contains prev_pt
+    ix: usize,
+    seg: Line,
+    prev: PathSeg,
+    next: PathSeg,
+}
+
+impl ErasureCandidate {
+    //https://github.com/googlefonts/glyphsLib/blob/74c63244fdb/Lib/glyphsLib/filters/eraseOpenCorners.py#L66-L71
+    fn points_are_right_of_line(&self) -> bool {
+        let prev_point = match self.prev {
+            PathSeg::Line(line) => line.start(),
+            PathSeg::Quad(quad_bez) => quad_bez.p1,
+            PathSeg::Cubic(cubic_bez) => cubic_bez.p2,
+        };
+        let next_point = match self.next {
+            PathSeg::Line(line) => line.end(),
+            PathSeg::Quad(quad_bez) => quad_bez.p1,
+            PathSeg::Cubic(cubic_bez) => cubic_bez.p1,
+        };
```

Comment:
At a glance it's hard to spot the difference between these matches. I suggest a trait implemented by PathSeg might be clearer, such that this would read

```rust
let prev_pt = self.prev.end_pt();
let next_pt = self.next.p1();
```

Reply:
the thing is it isn't the end point, because we are also including off-curves. This threw me too, at first. The other thing to keep in mind is that segments are independent, so they contain duplicate points (the end of one segment is the start of the subsequent one). I can add a comment to this effect, but since this logic is only used in one place it feels premature to try and do something fancier, here...

Reply:
Having it pop out in review as well suggests its worth a little polish imho. Bias to address review feedback basically.

Reply:
Oh I see it's the point before the end point ...what a weird process. self.prev.second_last_pt() or some such would be materially easier to read imho